core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java

```diff
@@ -3854,6 +3854,24 @@ private void checkLiteral2(String expression, String expected) {
     sql(expected).exec();
   }
 
+  @Test public void testCastInStringOperandOfComparison() {
+    final String query = "select \"employee_id\" "
+        + "from \"foodmart\".\"employee\" "
+        + "where 10 = cast('10' as int) and \"birth_date\" = cast('1914-02-02' as date) or "
+        + "\"hire_date\" = cast('1996-01-01 '||'00:00:00' as timestamp)";
+    final String expected = "SELECT \"employee_id\"\n"
+        + "FROM \"foodmart\".\"employee\"\n"
+        + "WHERE 10 = '10' AND \"birth_date\" = '1914-02-02' OR \"hire_date\" = '1996-01-01 ' || "
+        + "'00:00:00'";
+    final String expectedBiqquery = "SELECT employee_id\n"
+        + "FROM foodmart.employee\n"
+        + "WHERE 10 = CAST('10' AS INTEGER) AND birth_date = '1914-02-02' OR hire_date = "
```

Comment:
I am really to calcite's RelToSql. Just curious, for `CAST('10' AS INTEGER)`, it is correct on the `INTEGER`? As far as I know BigQuery only supports `INT64` for integer type (see [1])

[1] : https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#integer-type

Reply:
Ok. I think this comment is related to CALCITE-3381. The test case here is expected but it should be fixed by other JIRA.

Reply:
#1482 is ongoing.

Reply:
@soma-mondal. Because BigQuery does have `INTEGER` or INT32. It only supports INT64. 

The code leaves `INTEGER` unhandled because I wasn't sure if `INTEGER` should be cast to `INT64` or an exception should be thrown.

Reply:
@amaliujia  If BigQuery only has INT64 type, we should fix this, could you fire another issue and apply a patch ? You may need to override the `SqlDialect#getCastSpec` method.

Reply:
I see. I will send an email to discuss how to handle unsupported type for BigQuery and then send a patch.

Basically I will ask if people like to throw an exception directly or does a cast if that's compatible.